Makefile

```diff
@@ -33,43 +35,58 @@ run: generate lint manifests
 
 # Install CRDs into a cluster
 install: manifests
-	kustomize build config/crd | kubectl apply -f -
+	kubectl apply -f manifests/0000_31_cluster-baremetal-operator_02_metal3provisioning.crd.yaml
 
 # Uninstall CRDs from a cluster
 uninstall: manifests
-	kustomize build config/crd | kubectl delete -f -
+	kubectl delete -f manifests/0000_31_cluster-baremetal-operator_02_metal3provisioning.crd.yaml
 
 # Deploy controller in the configured Kubernetes cluster in ~/.kube/config
-deploy: manifests
-	cd config/cluster-baremetal-operator && kustomize edit set image controller=${IMG}
-	kustomize build config/default | kubectl apply -f -
+deploy: generate
+	cd config/cluster-baremetal-operator && $(KUSTOMIZE) edit set image controller=${IMG}
+	$(KUSTOMIZE) build config/profiles/$(MANIFEST_PROFILE) | kubectl apply -f -
+
+# this is to just get the order right
+RBAC_LIST = manifests/rbac.authorization.k8s.io_v1_role_cluster-baremetal-operator.yaml \
+	manifests/rbac.authorization.k8s.io_v1_clusterrole_cluster-baremetal-operator.yaml \
+	manifests/rbac.authorization.k8s.io_v1_rolebinding_cluster-baremetal-operator.yaml \
+  manifests/rbac.authorization.k8s.io_v1_clusterrolebinding_cluster-baremetal-operator.yaml
 
 # Generate manifests e.g. CRD, RBAC etc.
 .PHONY: manifests
-manifests:
-	hack/gen-crd.sh
+manifests: generate
+	$(KUSTOMIZE) build config/profiles/$(MANIFEST_PROFILE) -o manifests/
```

Comment:
My concern here is about generating directly the kustomize output into the `/manifests` folder. Whatever ends up here will be loaded by CVO during installation/upgrade. What about storing the intermediate generation into a temp dir, and then moving just the selected rendered artifacts into `/manifests`?

Reply:
ok, i can do that.